02-oop-inheritance/ex-complete/event_emiter.js

```diff
@@ -0,0 +1,32 @@
+export class EventEmitter{
+
+    constructor(){
+        this.listeners = [];
+    }
+
+    on(event, callback){
+        if(this.listeners[event]){
+            this.listeners[event].push(callback);
+        }
+        else{
+            this.listeners[event] = [callback];
+        }
+    }
+
+    emit(event, args){
+        let eventListeners = this.listeners[event];
+        if(eventListeners){
+            eventListeners.forEach(eventName => eventName(args));
+       }
+    }
+
+    off(event, callback){
+        let eventListeners = this.listeners[event];
+        if(this.listeners[event]){
+            let index = eventListeners.indexOf(callback)
```

Comment:
En estas tres lineas ademas del uso de let que es incorrecto (se deberia haber usado const), hay una cosa rara. Definis eventListenesrs a partir de this.listeners[event], para el if utilizas this.listeners[event] (habiendo ya creado la variable antes con ese mismo valor) y finalmente volves a usar la variable.